Add ChanUpgradeCancelConfirm datagram and handler to reset upgrade state (#722)
cosmos/ics: spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

### Cancel Upgrade Process

> **Sub-Definitions**:

Reviewer: I don't think it's necessary to have a subsection here for defining an aborting chain. You could just say directly something like this:
> During the upgrade handshake a chain may encounters an error and cancel the upgrade. The first channel to do so is referred to as the *aborting chain*. The aborting chain  may cancel the upgrade by writing an error receipt into the error path and restoring the original channel to `OPEN`. The counterparty must then restore its original channel to `OPEN` as well.

`Aborting Chain`: This is the chain that first encounters an error, writes data to the error receipt and restores the channel.

**Protocol**:

During the upgrade handshake an aborting chain may cancel the upgrade by writing an error receipt into the error path and restoring the original channel to `OPEN`. The counterparty must then restore its original channel to `OPEN` as well.

> A channelEnd may only cancel the upgrade during the upgrade negotiation process (`TRY`, `ACK`). Thus, the counterparty channel state must be either `UPGRADE_INIT` or `UPGRADE_TRY`. An upgrade cannot be cancelled on one end once the other chain has already completed its upgrade and moved to `OPEN` since that will lead the channel to being in an invalid state.

Reviewer: ```suggestion
The aborting chain may cancel the upgrade only during the upgrade negotiation process (`TRY`, `ACK`). Thus, the channel state on the counterparty chain must be either `UPGRADE_INIT` or `UPGRADE_TRY`. The aborting chain cannot cancel the channel upgrade once the counterparty chain has already completed its upgrade and moved to `OPEN`, since that will lead to the channel being in an invalid state.
```

> Once the aborting chain has written to the error receipt, a relayer must submit `ChanUpgradeCancelInit` to the counterparty chain:

Reviewer: ```suggestion
Once the aborting chain has written to the error receipt, a relayer must submit `ChanUpgradeCancelInitMsg` to the counterparty chain:
```
